mmv1/third_party/terraform/services/parametermanager/resource_parameter_manager_parameter_test.go.tmpl

```diff
@@ -106,4 +106,110 @@ resource "google_parameter_manager_parameter" "parameter-with-labels" {
 `, context)
 }
 
+func TestAccParameterManagerParameter_kmsKeyUpdate(t *testing.T) {
+	t.Parallel()
+	acctest.BootstrapIamMembers(t, []acctest.IamMember{
+		{
+			Member: "serviceAccount:service-{project_number}@gcp-sa-pm.iam.gserviceaccount.com",
+			Role:   "roles/cloudkms.cryptoKeyEncrypterDecrypter",
+		},
+	})
+
+	context := map[string]interface{}{
+		"kms_key":  acctest.BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", "global", "tf-parameter-manager-managed-1").CryptoKey.Name,
+		"kms_key_other":  acctest.BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", "global", "tf-parameter-manager-managed-2").CryptoKey.Name,
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t),
+		CheckDestroy:             testAccCheckParameterManagerParameterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccParameterManagerParameter_withoutkmsKey(context),
+			},
+			{
+				ResourceName:            "google_parameter_manager_parameter.parameter-with-kms-key",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "parameter_id", "terraform_labels"},
+			},
+			{
+				Config: testAccParameterManagerParameter_kmsKeyUpdate(context),
```

Comment:
update tests need to have a ConfigPlanChecks set on all Config steps after the first to verify that the resource under test is only being updated, not deleted and replaced. See https://googlecloudplatform.github.io/magic-modules/test/test/#add-an-update-test for more information.

```suggestion
				Config: testAccParameterManagerParameter_kmsKeyUpdate(context),
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("google_parameter_manager_parameter.parameter-with-kms-key", plancheck.ResourceActionUpdate),
					},
				},
```